### PR TITLE
Increase default font size and add serve_templates

### PR DESCRIPTION
**Pull request by rikhuijzer**
This PR is related to https://github.com/tlienart/FranklinTemplates.jl/issues/102. There, the default font size is the main point, but I also figured that a way serving all the templates would be convenient and came across some other stuff. The proposed changes are roughly as follows; the arguments for these changes are listed below.

- [x] Add `serve_templates`.
- [x] Document `serve_templates`.
- [x] Increase html font size to 19px.
- [x] Ensure that the increased html font size doesn't mess up relative elements such as `rem` dimensioned elements.
- [x] Decrease tables and code font size.
- [x] Increase toc spacing.
- [x] Use CSS variables for colors like ianstormtailor.com.

The output from this PR is at <https://rikhuijzer.github.io/FranklinTemplates.jl/>. [![][ci-img]][ci-url] 

## serve_templates
To be able to quickly see changes on all the templates, this PR adds `serve_templates()`. It can be used via
```
$ cd FranklinTemplates.jl

$ julia
 
pkg> dev .

julia> using FranklinTemplates; FranklinTemplates.serve_templates()
```

## Increase html font size
As discussed in https://github.com/tlienart/FranklinTemplates.jl/issues/102: it is a good idea to bump the default font size up slightly to allow for easier reading on desktop and mobile. The font size is bumped from `15/16px` to `19px`. To keep it simple, no distinction is made between devices for the font size.

## Decrease tables and code font size
Size of elements can be used to signal what is important on the page and what is not. I would argue that for most of Franklin's users, tables and code are not as important as the main text. As a second argument, reading long texts can be very tiring and having a bigger font size reduces this. For code and tables, users typically don't have to read all the code and tables line by line and these elements cannot wrap around on small screens.  So, it makes sense to move the font size for code and tables down to `14px`. This is big enough for mobile readers according to Google Search Console. I have tried `12px` as well, but that resulted in a Mobile Usability error. So, assuming that Google bots know what they're doing, lets do `14px`. 

## Increase toc spacing
The defaults for the table of contents were too small, so I bumped it from `0.3rem 0` to `0.6 rem 0`. This fixes Google Search Console error "Clickable elements too close together" for mobile. This change is also supported by [double your whitespace](https://learnui.design/blog/7-rules-for-creating-gorgeous-ui-part-1.html#rule-3-double-your-whitespace).

[ci-img]: https://github.com/rikhuijzer/FranklinTemplates/workflows/deploy/badge.svg
[ci-url]: https://github.com/rikhuijzer/FranklinTemplates/actions

### CODE REVIEW

**Comment by rikhuijzer** [comment]
The [Work in progress] is removed from the title because GitHub clearly shows the number of items completed. Currently, this PR is still work in progress; let me know if you have any feedback already.

**Comment by rikhuijzer** [comment]
All templates should now look fairly similar to how they used to look. In the background, I've tried to increase the consistency and readability of the CSS. Also, I've tried to allow users to be able to easily make changes. This is done via using CSS variables and by moving the closing curly brackets in `franklin.css` to a new line.

**Comment by tlienart** [comment]
Thanks so much Rik!! I'll try to review this asap

**Comment by rikhuijzer** [comment]
If this PR is merged, it might be a good idea to use `npx stylelint --fix "**/*.css"`. This is a (very small) part of the diff

```
diff --git a/src/templates/basic/_css/basic.css b/src/templates/basic/_css/basic.css
index 1612a86..3b2ffc5 100644
--- a/src/templates/basic/_css/basic.css
+++ b/src/templates/basic/_css/basic.css
@@ -3,155 +3,188 @@
 ================================================================== */

 nav {
-    width: 64%;
-    display: inline-block; }
+  width: 64%;
+  display: inline-block;
+}

 nav ul {
-    padding-left: 0;
-    margin-top: 0;
-    margin-bottom: 0; }
+  padding-left: 0;
+  margin-top: 0;
+  margin-bottom: 0;
+}

 nav li {
-    display: inline-block; }
+  display: inline-block;
+}
```

File: Project.toml [hunk 1]
```diff
@@ -3,5 +3,8 @@ uuid = "3a985190-f512-4703-8d38-2a7944ed5916"
 authors = ["Thibaut Lienart <tlienart@me.com>"]
 version = "0.8.2"
 
+[deps]
+LiveServer = "16fef848-5104-11e9-1b77-fb7a48bbb589"
```
Comment (tlienart) [inline Project.toml:7]: This should be removed (not  used  in the package)

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -23,33 +23,15 @@ jobs:
...
-                    htmls = read(path, String);
-                    htmls = replace(htmls, "href=\"/" => "href=\"/FranklinTemplates.jl/");
-                    htmls = replace(htmls, "src=\"/" => "src=\"/FranklinTemplates.jl/");
-                    write(path, htmls);
-                end
-            end'
-    - name: Build and Deploy
+            using FranklinTemplates;
+            FranklinTemplates.build_templates()'
+    - name: Deploy
```
Comment (tlienart) [inline .github/workflows/deploy.yml:34]: Thanks for the cleanup!

**Review by tlienart — APPROVED** [review]

**Comment by tlienart** [comment]
This is phenomenal, thanks a lot! some of the text looks a bit huge to me now but I guess  this is a matter  of taste and I  like the use of variables (now 95% supported according to caniuse; it was less last I checked).  

I'm not against you running the linter on everything (I'll give you collaborator access for this so that you can directly merge such changes, no need for review). 

Thanks again so much! glad to have someone finally having done some good work on the templates 😄

**Pull request merged**